stew/objects.nim

```diff
@@ -59,13 +59,13 @@ when not compiles(len((1, 2))):
 # Get an object's base type, as a cstring. Ref objects will have an ":ObjectType"
 # suffix.
 # From: https://gist.github.com/stefantalpalaru/82dc71bb547d6f9178b916e3ed5b527d
-proc baseType*(obj: RootObj): cstring =
+proc baseType*(obj: RootObj): cstring {.deprecated.} =
```

Comment:
I think for these orc-breaking things, we should add a `when not orc:` in addition to the deprecation so that there's a hard compile error where it's actually broken

Reply:
> I'm pretty sure a macro could do it, if we need it

It is quite unsafe because there is no guarantee regarding the implementation details. Since https://github.com/nim-lang/Nim/pull/20789, the name becomes shorter for ARC/ORC and no longer generates type names for parent objects.